vortex-scalar/src/utf8.rs

```diff
@@ -7,19 +7,29 @@ use vortex_error::{vortex_bail, vortex_err, VortexError, VortexExpect as _, Vort
 
 use crate::{InnerScalarValue, Scalar, ScalarValue};
 
-#[derive(Debug, Hash, PartialEq, Eq)]
+#[derive(Debug, Hash)]
 pub struct Utf8Scalar<'a> {
```

Comment:
Can we make the scalars a TypedScalar<'a, T>? So we can share these, definitions

Reply:
Not right now